core/controllers/topic_viewer.py

```diff
@@ -67,6 +67,7 @@ def get(self, topic_name):
 
         canonical_story_dicts = []
         for story_summary in canonical_story_summaries:
+            all_nodes = story_fetchers.get_all_nodes_in_story(story_summary.id)
```

Comment:
Can't you perhaps modify/repurpose `get_pending_nodes_in_story` for this? Since both that and this function does the same database call (get_story_by_id) which seems unnecessary.

You only really need completed titles and all_dicts. I think both this can be returned from a single new function than multiple functions here. What do you think?

Reply:
Hi @aks681 , sorry for the late response. Should I return both pending and all nodes from ```get_pending_nodes_in_story``` and use whichever required out of the two?

Reply:
@aks681 nudge

Reply:
Yup, that sounds good! Though, you might need to rename the function for it.

Reply:
@aks681 Done, PTAL!